lib/kubernetes-deploy/runner.rb

```diff
@@ -106,6 +108,13 @@ def run
       phase_heading("Checking initial resource statuses")
       resources.each(&:sync)
 
+      ejson = EjsonSecretProvisioner.new(namespace: @namespace, context: @context, template_dir: @template_dir)
+      if ejson.secrets_requested?
```

Comment:
I find this a bit odd,  why do we have to create the object, ask it `hey do you want to do things?` and then call methods on it? instead of encapsulating all it in the `ejson` object.

```
# in this file just:
ejson.run()

# in the object

def run 
 return unless secrets_requested?
  ejson.create_secrets
  ejson.prune_managed_secrets
 end
end
```